posthog/hogql_queries/insights/trends/test/test_data_warehouse_query_builder.py

```diff
@@ -0,0 +1,221 @@
+from datetime import datetime
+from freezegun import freeze_time
+from posthog.hogql.modifiers import create_default_modifiers_for_team
+
+from posthog.hogql.query import execute_hogql_query
+from posthog.hogql.timings import HogQLTimings
+from posthog.hogql_queries.insights.trends.data_warehouse_trends_query_builder import DataWarehouseTrendsQueryBuilder
+from posthog.hogql_queries.utils.query_date_range import QueryDateRange
+from posthog.schema import (
+    BreakdownFilter,
+    BreakdownType,
+    DateRange,
+    DataWarehouseNode,
+    TrendsQuery,
+)
+from posthog.test.base import BaseTest
+from posthog.warehouse.models import DataWarehouseTable, DataWarehouseCredential
+
+from boto3 import resource
+from botocore.config import Config
+from posthog.settings import (
+    OBJECT_STORAGE_ACCESS_KEY_ID,
+    OBJECT_STORAGE_BUCKET,
+    OBJECT_STORAGE_ENDPOINT,
+    OBJECT_STORAGE_SECRET_ACCESS_KEY,
+)
+import s3fs
+from pyarrow import parquet as pq
+import pyarrow as pa
+
+from posthog.test.base import (
+    ClickhouseTestMixin,
+    snapshot_clickhouse_queries,
+)
+from posthog.hogql_queries.legacy_compatibility.filter_to_query import (
+    clean_entity_properties,
+)
+
+TEST_BUCKET = "test_storage_bucket-posthog.hogql.datawarehouse.trendquery"
+
+
+class TestDataWarehouseQueryBuilder(ClickhouseTestMixin, BaseTest):
+    def teardown_method(self, method) -> None:
+        s3 = resource(
+            "s3",
+            endpoint_url=OBJECT_STORAGE_ENDPOINT,
+            aws_access_key_id=OBJECT_STORAGE_ACCESS_KEY_ID,
+            aws_secret_access_key=OBJECT_STORAGE_SECRET_ACCESS_KEY,
+            config=Config(signature_version="s3v4"),
+            region_name="us-east-1",
+        )
+        bucket = s3.Bucket(OBJECT_STORAGE_BUCKET)
+        bucket.objects.filter(Prefix=TEST_BUCKET).delete()
+
+    def get_response(self, trends_query: TrendsQuery):
+        query_date_range = QueryDateRange(
+            date_range=trends_query.dateRange,
+            team=self.team,
+            interval=trends_query.interval,
+            now=datetime.now(),
+        )
+
+        timings = HogQLTimings()
+        modifiers = create_default_modifiers_for_team(self.team)
+
+        query_builder = DataWarehouseTrendsQueryBuilder(
+            trends_query=trends_query,
+            team=self.team,
+            query_date_range=query_date_range,
+            series=trends_query.series[0],
+            timings=timings,
+            modifiers=modifiers,
+        )
+
+        query = query_builder.build_query()
+
+        return execute_hogql_query(
+            query_type="TrendsQuery",
+            query=query,
+            team=self.team,
+            timings=timings,
+        )
+
+    def create_parquet_file(self):
```

Comment:
We'll likely need to add some higher-level helper funcs somewhere in our testing framework for doing this kinda stuff. I imagine I'm gonna need the same when doing the table linking too